test/e2e/self_hosted_test.go

```diff
@@ -24,19 +24,6 @@ import (
 	"k8s.io/utils/ptr"
 )
 
```

Comment:
The coverage this test provides is a subset of `When testing Cluster API working on self-hosted clusters using ClusterClass [ClusterClass]`